RELEASE/scripts/autoscend/quests/level_11.ash

```diff
@@ -562,112 +568,83 @@ boolean L11_mcmuffinDiary()
-	if(in_zelda() && get_property("desertExploration").to_int() < 100)
+	if(in_koe())
 	{
-		string discard = visit_url("place.php?whichplace=desertbeach");
+		return false;		//impossible to get compass in this path. [The Shore, Inc] is unavailable
 	}
-
-	if(get_property("desertExploration").to_int() >= 100)
+	if(auto_my_path() == "Way of the Surprising Fist" || $class[Avatar of Boris] == my_class())
 	{
-		return false;
+		return false;		//cannot equip offhand item in these paths
 	}
 
-	item desertBuff = $item[none];
-	int progress = 1;
-	if(possessEquipment($item[UV-resistant compass]))
+	if(item_amount($item[Shore Inc. Ship Trip Scrip]) == 0)
 	{
-		desertBuff = $item[UV-resistant compass];
-		progress = 2;
+		return LX_doVacation();
 	}
-	if(possessEquipment($item[Ornate Dowsing Rod]) && is_unrestricted($item[Hibernating Grimstone Golem]))
+	
+	if(create(1, $item[UV-Resistant Compass]))
 	{
-		desertBuff = $item[Ornate Dowsing Rod];
-		progress = 3;
+		return true;
 	}
-	if((get_property("bondDesert").to_boolean()) && ($location[The Arid\, Extra-Dry Desert].turns_spent > 0))
+	else
 	{
-		progress += 2;
+		cli_execute("refresh inv");
+		if(possessEquipment($item[UV-resistant compass]))
+		{
+			return true;
+		}
+		else abort("I have the Scrip for it but am failing to buy [UV-resistant compass] for some reason. buy it manually and run me again");
 	}
 
-	boolean failDesert = true;
-	if(possessEquipment(desertBuff))
+	return false;
+}
+
+boolean L11_aridDesert()
+{
+	if(internalQuestStatus("questL11Desert") != 0)
 	{
-		failDesert = false;
+		return false;
 	}
-	if($classes[Avatar of Boris, Avatar of Sneaky Pete] contains my_class())
+
+	// Fix broken desert tracking. pocket familiars failing as of r19010. plumber as of r20019
+	if(in_zelda() || auto_my_path() == "Pocket Familiars")
 	{
-		failDesert = false;
+		visit_url("place.php?whichplace=desertbeach", false);
 	}
-	if(auto_my_path() == "Way of the Surprising Fist")
+	if(get_property("desertExploration").to_int() >= 100)
 	{
-		failDesert = false;
+		return false;		//done exploring
 	}
-	if(get_property("bondDesert").to_boolean())
+	
+	if(LX_ornateDowsingRod(true)) return true;		//spend adv trying to get [Ornate Dowsing Rod]. doing_desert_now = true.
+	if(L11_getUVCompass()) return true;				//spend adv trying to get [UV-resistant compass]
+
+	item desertBuff = $item[none];
+	int progress = 1;
+	if(possessEquipment($item[UV-resistant compass]))
 	{
-		failDesert = false;
+		desertBuff = $item[UV-resistant compass];
+		progress = 2;
 	}
-	if(in_koe())
+	if(possessEquipment($item[Ornate Dowsing Rod]) && is_unrestricted($item[Ornate Dowsing Rod]))
 	{
-		failDesert = false;
+		desertBuff = $item[Ornate Dowsing Rod];
+		progress = 3;
 	}
-
-	if(failDesert)
+	if((get_property("bondDesert").to_boolean()) && ($location[The Arid\, Extra-Dry Desert].turns_spent > 0))
```

Comment:
AoSP's Blacklight Bulb works similarly. Might want to add a todo comment if there's no mafia support for it yet though (I haven't checked).